src/nvc.rs

```diff
@@ -155,10 +257,19 @@ impl Nvc {
         self.psw_interrupt_mask_level = ((value as usize) >> 16) & 0x0f;
     }
 
+    pub fn fetch_instruction(&mut self, interconnect: &mut Interconnect, addr: u32) -> (u16, bool) {
+        let halfword = interconnect.read_halfword(addr);
```

Comment:
I wonder if it would be more clear if the actual lookup were performed _by_ the cache, instead of by the cpu in this case and then checking if that would've used the cache or not. We could have a `read_halfword` fn (or similar) on the `NvcCache` struct that would take in a `&mut Interconnect` and return basically the same tuple that's found here.